### PR TITLE
User and Ticket CRUD functionality

### PR DESCRIPTION
**Pull request by sanabriadiego**
Added CRUD functionality for each table.

### CODE REVIEW

File: src/services/service.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
+import { QueryResult } from 'pg'
+const client = require('../postgres/conn')
+client.connect()
+
```
Comment (SHARON-CALCINA) [inline src/services/service.ts:4]: delete blank rows in the whole file and use ES6 or ES5 to import modules

File: src/postgres/conn.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import 'dotenv/config'
+
```
Comment (SHARON-CALCINA) [inline src/postgres/conn.ts]: delete blank spaces and use ES6 or ES5 to import modules

File: src/index.ts [hunk 3]
```diff
@@ -1,37 +1,24 @@
...
-});
-
-// Starting the Server
-app.listen(app.get('port'), () => {
-    console.log(`Server on port`, app.get('port'));
+import express from 'express'
+import 'dotenv/config'
+import { QueryResult } from 'pg'
+const userServices = require('./services/service')
+
```
Comment (SHARON-CALCINA) [inline src/index.ts]: delete blank spaces and use ES6 or ES5 to import modules

File: src/index.ts [hunk 4]
```diff
@@ -1,37 +1,24 @@
...
+// Initializations
+const app = express()
+const PORT = process.env.SERVER_PORT || 4001
+
+// middlewares
```
Comment (SHARON-CALCINA) [inline src/index.ts]: erase middlewares comment and blank rows

File: package.json [hunk 5]
```diff
@@ -1,27 +1,31 @@
...
+    "@types/pg": "^8.6.5",
+    "nodemon": "^1.18.9",
+    "ts-node": "^10.9.1",
+    "typescript": "^3.9.10"
+  },
+  "dependencies": {
+    "@types/typescript": "^2.0.0",
+    "dotenv": "^16.0.3",
+    "express": "^4.18.2",
+    "mongoose": "^6.6.6",
```
Comment (SHARON-CALCINA) [inline package.json]: mongo db is not needed

File: src/services/service.ts [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+            // res.status(200).json(result['rows'])
+            res.status(200).json(result)
+        })
+    } catch (e) {
+        console.log(e)
+        throw (e)
+    }
+
+}
+declare namespace Express {
```
Comment (SHARON-CALCINA) [inline src/services/service.ts]: erase this repeated function and put the interface in some other file so this file is just for services

File: src/services/service.ts [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+
+//GET
+
+const getUsers = async (_req: Express.Request, res: Express.Response) => {
+    try {
+        await client.query('SELECT * from users', (error: Error, result: QueryResult) => {
+            if (error) {
+                throw error
+            }
+            // res.status(200).json(result['rows'])
```
Comment (SHARON-CALCINA) [inline src/services/service.ts]: erase commented row

File: src/services/service.ts [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+    }
+}
+
+const getTicket = async (_req: Express.Request, res: Express.Response) => {
+    try {
+        await client.query('SELECT * from ticket', (error: Error, result: QueryResult) => {
+            if (error) {
+                throw error
+            }
+            // res.status(200).json(result['rows'])
```
Comment (SHARON-CALCINA) [inline src/services/service.ts]: erase commented code row

File: src/services/service.ts [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+        status: any;
+    }
+}
+
+//POST
+
+const createUser = async (request: Express.Request, response: Express.Response) => {
+    const { name, email, cellphone, age, address, created, country, state } = request.body
+    try {
+        await client.query('INSERT INTO users (name, email, cellphone, age, address, created, country, state) VALUES ($1, $2, $3, $4, $5, $6, $7, $8) RETURNING *', [name, email, cellphone, age, address, created, country, state], (error: Error, results: QueryResult) => {
```
Comment (SHARON-CALCINA) [inline src/services/service.ts]: add validations to verify missed (not null) fields before the query is sent  for CREATEUSER & CREATETICKET function

**Review by SHARON-CALCINA — DISMISSED** [review]

File: src/services/service.ts (hunk 7 above)
Comment (walterr2345) [inline src/services/service.ts]: please if this comment is necessary?

File: src/services/service.ts (hunk 8 above)
Comment (walterr2345) [inline src/services/service.ts]: same here

File: src/index.ts [hunk 10]
```diff
@@ -1,37 +1,24 @@
...
-
-});
-
-// Starting the Server
-app.listen(app.get('port'), () => {
-    console.log(`Server on port`, app.get('port'));
+import express from 'express'
+import 'dotenv/config'
+import { QueryResult } from 'pg'
+const userServices = require('./services/service')
```
Comment (annieccr) [inline src/index.ts]: Standarize imports

File: src/services/service.ts [hunk 11]
```diff
@@ -0,0 +1,167 @@
+import { QueryResult } from 'pg'
+const client = require('../postgres/conn')
+client.connect()
+
+//GET
+
+const getUsers = async (_req: Express.Request, res: Express.Response) => {
+    try {
+        await client.query('SELECT * from users', (error: Error, result: QueryResult) => {
```
Comment (annieccr) [inline src/services/service.ts]: It isn't necessary the callback because you are using try catch and async await

File: src/services/service.ts [hunk 12]
```diff
@@ -0,0 +1,167 @@
...
+        body: any;
+        params: any;
+    }
+    export interface Response {
+        status: any;
+    }
+}
+
+const getTicket = async (_req: Express.Request, res: Express.Response) => {
+    try {
```
Comment (annieccr) [inline src/services/service.ts:32]: Review try catch and callbacks

File: src/services/service.ts [hunk 13]
```diff
@@ -0,0 +1,167 @@
...
+        })
+    } catch (e) {
+        console.log(e)
+        throw (e)
+    }
+
+}
+
+const createTicket = async (request: Express.Request, response: Express.Response) => {
+    const { price, category, match_date, stadium, created } = request.body
```
Comment (annieccr) [inline src/services/service.ts:62]: Review if some columns are required

File: src/services/service.ts [hunk 14]
```diff
@@ -0,0 +1,167 @@
...
+        console.log(e)
+        throw (e)
+    }
+
+}
+
+//PUT
+
+const updateUser = async (request: Express.Request, response: Express.Response) => {
+    const id = parseInt(request.params.id)
```
Comment (annieccr) [inline src/services/service.ts:94]: Verify if ID isn't null or empty

File: src/services/service.ts [hunk 15]
```diff
@@ -0,0 +1,167 @@
...
+    const { name, email, cellphone, age, address, country, state } = request.body
+    try {
+        await client.query('UPDATE users SET name = $1, email = $2, cellphone = $3, age= $4, address = $5, country = $6, state = $7 WHERE user_id = $8', [name, email, cellphone, age, address, country, state, id], (error: Error, results: QueryResult) => {
+            if (error) {
```
Comment (annieccr) [inline src/services/service.ts:82]: The try catch already catches the error

File: src/services/service.ts [hunk 16]
```diff
@@ -0,0 +1,167 @@
...
+        }
+        )
+    } catch (e) {
+        console.log(e)
+        throw (e)
+    }
+}
+
+const updateTicket = async (request: Express.Request, response: Express.Response) => {
+    const id = parseInt(request.params.id)
```
Comment (annieccr) [inline src/services/service.ts:111]: Verify if the ID isn't null or empty

File: src/services/service.ts [hunk 17]
```diff
@@ -0,0 +1,167 @@
...
+    const { price, category, match_date, stadium, created } = request.body
+    try {
+        await client.query('UPDATE ticket SET price = $1, category = $2, match_date = $3, stadium= $4, created = $5 WHERE ticket_id = $6', [price, category, match_date, stadium, created, id], (error: Error, results: QueryResult) => {
+            if (error) {
```
Comment (annieccr) [inline src/services/service.ts:98]: The try catch already catches the error

File: src/services/service.ts [hunk 18]
```diff
@@ -0,0 +1,167 @@
...
+    }
+}
+
+//DELETE
+
+const deleteUser = (request: Express.Request, response: Express.Response) => {
+    const id = parseInt(request.params.id)
+    try {
+        client.query('DELETE FROM users WHERE user_id = $1', [id], (error: Error, results: QueryResult) => {
+            if (error) {
```
Comment (annieccr) [inline src/services/service.ts:114]: The try catch already catches the error

File: src/services/service.ts [hunk 19]
```diff
@@ -0,0 +1,167 @@
...
+        console.log(e)
+        throw (e)
+    }
+}
+
+const deleteTicket = (request: Express.Request, response: Express.Response) => {
+    const id = parseInt(request.params.id)
+    try {
+        client.query('DELETE FROM ticket WHERE ticket_id = $1', [id], (error: Error, results: QueryResult) => {
+            if (error) {
```
Comment (annieccr) [inline src/services/service.ts:129]: The try catch already catches the error

File: src/services/service.ts [hunk 20]
```diff
@@ -0,0 +1,167 @@
...
+    } catch (e) {
+        console.log(e)
+        throw (e)
+    }
+}
+
+//DELETE
+
+const deleteUser = (request: Express.Request, response: Express.Response) => {
+    const id = parseInt(request.params.id)
```
Comment (annieccr) [inline src/services/service.ts:130]: verify if the ID isn't null or empty

**Review by annieccr — COMMENTED** [review]
reviewed

File: src/services/service.ts [hunk 21]
```diff
@@ -0,0 +1,167 @@
...
+const deleteTicket = (request: Express.Request, response: Express.Response) => {
+    const id = parseInt(request.params.id)
+    try {
+        client.query('DELETE FROM ticket WHERE ticket_id = $1', [id], (error: Error, results: QueryResult) => {
+            if (error) {
+                throw error
+            }
+            response.status(200).send(`User deleted with ID: ${id}`)
+        })
+    } catch (e) {
```
Comment (walterr2345) [inline src/services/service.ts]: maybe for more legible e can change for error

File: src/services/service.ts [hunk 22]
```diff
@@ -0,0 +1,167 @@
...
+const deleteUser = (request: Express.Request, response: Express.Response) => {
+    const id = parseInt(request.params.id)
+    try {
+        client.query('DELETE FROM users WHERE user_id = $1', [id], (error: Error, results: QueryResult) => {
+            if (error) {
+                throw error
+            }
+            response.status(200).send(`User deleted with ID: ${id}`)
+        })
+    } catch (e) {
```
Comment (walterr2345) [inline src/services/service.ts]: maybe for more legible e can change for error

File: src/services/service.ts [hunk 23]
```diff
@@ -0,0 +1,167 @@
...
+        throw (e)
+    }
+
+}
+
+const createTicket = async (request: Express.Request, response: Express.Response) => {
+    const { price, category, match_date, stadium, created } = request.body
+    try {
+        await client.query('INSERT INTO ticket ( price, category, match_date, stadium, created) VALUES ($1, $2, $3, $4, $5) RETURNING *', [ price, category, match_date, stadium, created], (error: Error, results: QueryResult) => {
+            if (error) {
```
Comment (walterr2345) [inline src/services/service.ts:65]: validacion if one field is empty

File: src/services/service.ts [hunk 24]
```diff
@@ -0,0 +1,167 @@
...
+const getUsers = async (_req: Express.Request, res: Express.Response) => {
+    try {
+        await client.query('SELECT * from users', (error: Error, result: QueryResult) => {
+            if (error) {
+                throw error
+            }
+            // res.status(200).json(result['rows'])
+            res.status(200).json(result)
+        })
+    } catch (e) {
```
Comment (walterr2345) [inline src/services/service.ts:24]: maybe for more legible e can change for error

File: src/services/service.ts (hunk 1 above)
Comment (sanabriadiego) [reply src/services/service.ts:4]: done

File: src/postgres/conn.ts (hunk 2 above)
Comment (sanabriadiego) [reply src/postgres/conn.ts]: done

File: src/index.ts (hunk 3 above)
Comment (sanabriadiego) [reply src/index.ts]: done

File: src/index.ts (hunk 4 above)
Comment (sanabriadiego) [reply src/index.ts]: done

File: package.json (hunk 5 above)
Comment (sanabriadiego) [reply package.json]: done

File: src/services/service.ts (hunk 6 above)
Comment (sanabriadiego) [reply src/services/service.ts]: done

File: src/services/service.ts (hunk 7 above)
Comment (sanabriadiego) [reply src/services/service.ts]: done

File: src/services/service.ts (hunk 8 above)
Comment (sanabriadiego) [reply src/services/service.ts]: done

File: src/services/service.ts (hunk 9 above)
Comment (sanabriadiego) [reply src/services/service.ts]: done

File: src/services/service.ts (hunk 7 above)
Comment (sanabriadiego) [reply src/services/service.ts]: done

File: src/services/service.ts (hunk 8 above)
Comment (sanabriadiego) [reply src/services/service.ts]: done

File: src/index.ts (hunk 10 above)
Comment (sanabriadiego) [reply src/index.ts]: done

File: src/services/service.ts (hunk 11 above)
Comment (sanabriadiego) [reply src/services/service.ts]: done

File: src/services/service.ts (hunk 12 above)
Comment (sanabriadiego) [reply src/services/service.ts:32]: done

File: src/services/service.ts (hunk 13 above)
Comment (sanabriadiego) [reply src/services/service.ts:62]: done

File: src/services/service.ts (hunk 15 above)
Comment (sanabriadiego) [reply src/services/service.ts:82]: done

File: src/services/service.ts (hunk 17 above)
Comment (sanabriadiego) [reply src/services/service.ts:98]: done

File: src/services/service.ts (hunk 18 above)
Comment (sanabriadiego) [reply src/services/service.ts:114]: done

File: src/services/service.ts (hunk 19 above)
Comment (sanabriadiego) [reply src/services/service.ts:129]: done

File: src/services/service.ts (hunk 21 above)
Comment (sanabriadiego) [reply src/services/service.ts]: done

File: src/services/service.ts (hunk 22 above)
Comment (sanabriadiego) [reply src/services/service.ts]: done

File: src/services/service.ts (hunk 24 above)
Comment (sanabriadiego) [reply src/services/service.ts:24]: done

**Pull request merged**